logstash-core-event-java/src/main/java/com/logstash/Javafier.java

```diff
@@ -127,6 +134,16 @@ public static Object deep(IRubyObject o) {
             case RubyNil: return deep((RubyNil)o);
             case True: return deep((RubyBoolean.True)o);
             case False: return deep((RubyBoolean.False)o);
+            case MapJavaProxy: return deep(((MapJavaProxy) o).to_hash());
+            case ArrayJavaProxy:
+            case ConcreteJavaProxy:
+                Object obj = JavaUtil.unwrapJavaObject(o);
```

Comment:
I thought about what you said regarding using a method here and I think we should.
I suggest that we raise an error rather than fall through when we get a ConcreteJavaProxy value that we have not coded for. WDYT?